### PR TITLE
make migrated node has higher candidate priority

### PR DESCRIPTION
**Pull request by cnderrauber**
will merge after @davidzhao 's pr

### CODE REVIEW

File: pkg/rtc/participant.go [hunk 1]
```diff
@@ -589,6 +603,9 @@ func (p *ParticipantImpl) SetMigrateState(s types.MigrateState) {
 		p.pendingOffer = nil
 		// in case of migration, subscriber data channel will not fire OnOpen callback
 		p.activeCounter.CAS(0, 2)
+
+		// use lower network cost to make sure migrated's node has higher priority candidate
+		p.networkCost--
```
Comment (davidzhao) [inline pkg/rtc/participant.go:608]: how would this new network cost transfer to the new node?

Comment (boks1971) [reply pkg/rtc/participant.go:608]: Also where is this used @cnderrauber . Sorry not able to see it as it is combined with the other large PR.

Comment (cnderrauber) [reply pkg/rtc/participant.go:608]: @davidzhao we don't need transfer it to the new node, the higher priority candidate type (peer reflex) will only happen in first connect in our service env(happened when stun server no response or too slow than sfu node). Condition of peer reflex is ice agent got a XOR-MAPPED-ADDRESS that not match to any known address, then a local candidate promote to a peer reflex candidate.

@boks1971 Update now, the code change should be clear.

Comment (davidzhao) [reply pkg/rtc/participant.go:608]: what happens to the second migration? Maybe I'm not understanding the root cause.. but now you'll end up with ICE candidates from both new and old nodes at 99?

File: pkg/rtc/participant_signal.go [hunk 2]
```diff
@@ -178,6 +178,7 @@ func (p *ParticipantImpl) sendIceCandidate(c *webrtc.ICECandidate, target liveki
 	// write candidate
 	p.params.Logger.Debugw("sending ice candidates",
 		"candidate", c.String(), "target", target)
+	ci.Candidate += fmt.Sprintf(" network-cost %d", p.networkCost)
```
Comment (boks1971) [inline pkg/rtc/participant_signal.go:181]: Does this work on both Chrome and Safari? Wondering if instead hacking the priority directly would work better? But knowing the priority will be harder.

File: pkg/rtc/participant.go (hunk 1 above)
Comment (cnderrauber) [reply pkg/rtc/participant.go:608]: yeah, they have same cost 99, and new node have new generation, then will fallback to normal case that no problem in other region like sgp, will successed.

File: pkg/rtc/participant_signal.go (hunk 2 above)
Comment (cnderrauber) [reply pkg/rtc/participant_signal.go:181]: yeah, hacking candidate is hard that we should make sure server side's priority less than client's (min priority takes high bits in priority calc), and perform all priority calculate to make sure new node's candidate to be used, and  the client's peer reflex candidate's priority is hard to known too.

**Review by davidzhao — APPROVED** [review]
confirmed it does not break ICE restart for Safari and Chrome.

**Pull request merged**